torchgen/api/types.py

```diff
@@ -293,121 +234,6 @@ def remove_const_ref(self) -> "CType":
         return self
```

Comment:
If you plan to refactor CType to be inheritance based, could you split that out into a separate PR? (I'm not entirely sure how it helps though...)

Reply:
OK, it looks like it's because you want to add custom CTypes

Reply:
Yeah obviously I will split this into several PRs. This one is a proof of concept so that we can align on the direction.